### ISSUE TITLE
onChange handler returns updated values in oldFlags object

### ISSUE THREAD

**Issue by harenlewis** [BulletTrainHQ/bullet-train-js-client#25]
When I try to access the oldFlags object to check for state updates I get updated latest values.

```
onChange: (oldFlags, params) => {
  const oldLink = oldFlags["dh_web_link"].value;  // test.com
  const newLink = bulletTrain.getValue("dh_web_link");  // test.com
}
```
Expected behavior:
I should receive the old value when I access oldLink

Is this something to do with the recent oldFlags fix #22 ?

**Comment by kyle-ssg** [comment]
Hey @harenlewis, going to schedule some time Friday to sort this. I’m sure it’ll be quite obvious when I look at it.

Ofcourse if you want to look at doing a pr in the meantime that would be amazing!

**Comment by harenlewis** [comment]
@kyle-ssg  Hey thanks for the reply. Sure I've gone through the code, can you help me setup the development env, I mean how to get the library running for development.
Or if there's any doc that I can refer to?

**Comment by kyle-ssg** [comment]
@harenlewis sorry for the delay, a few things.

I've deployed version 0.2.0 on bullet-train-js-client and react-native-bullet-train to resolve this issue.

- I've added instructions on developing the client library locally.
- You're correct the associated PR did break this, however old flags needed to be immutable. The above code snippet will now work.
- params now contains a flagsChanged and traitsChanged boolean which is based on a deep equals of before and after.

If you have any questions or comments just let me know and I'll reopen this!
